src/app/onboarding/roles/page.tsx

```diff
@@ -1,57 +1,178 @@
+      'LEGAL_FELLOW,INTERPRETER',
+      '',
+    ])
+    .superRefine((input, ctx) => {
+      if (input === '')
+        ctx.addIssue({
+          code: 'custom',
+          message: 'Must include at least one role',
+        });
+      return ctx;
+    }),
+});
+
 export default function Page() {
   const onboarding = useContext(OnboardingContext);
+  const { push } = useRouter();
+
+  const onSubmit = (values: z.infer<typeof roleSchema>) => {
+    if (!onboarding) throw new Error('Fatal: no onboarding layout detected');
 
-  const setRoles = (roles: Set<RoleEnum>) => {
-    onboarding?.setRoles(roles);
-    if (roles.size !== 0) {
-      onboarding?.setCanContinue(true);
+    const oldRoles = onboarding.roles;
+    const roles = values.roles.split(',') as RoleEnum[];
+    onboarding.setRoles(roles);
+
+    let newFlow: FlowData[];
+    const isAttorney = roles.includes('ATTORNEY');
+    const isLegalFellow = roles.includes('LEGAL_FELLOW');
+
+    if (isAttorney) {
+      newFlow = ATTORNEY_FLOW;
+    } else if (isLegalFellow) {
+      newFlow = LEGAL_FELLOW_FLOW;
     } else {
-      onboarding?.setProgress(0);
-      onboarding?.setCanContinue(false);
+      newFlow = INTERPRETER_FLOW;
     }
 
-    if (roles.has('ATTORNEY')) {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Legal Experience', url: 'legal-experience' },
-        { name: 'Done', url: 'done' },
-      ]);
-    } else {
-      onboarding?.setFlow([
-        { name: 'Roles', url: 'roles' },
-        { name: 'Basic Info', url: 'basic-information' },
-        { name: 'Availability', url: 'availability' },
-        { name: 'Done', url: 'done' },
+    // remove role-specific data if role changes
+    if (oldRoles.includes('ATTORNEY') && !isAttorney)
+      onboarding.removeFromProfile([
+        'bar_number',
+        'eoir_registered',
+        'state_barred',
       ]);
-    }
+
+    if (oldRoles.includes('LEGAL_FELLOW') && !isLegalFellow)
+      onboarding.removeFromProfile(['expected_bar_date', 'eoir_registered']);
+
+    // cap progress to 3 (legal info)
+    onboarding.setProgress(progress => Math.min(progress, 3));
+
+    onboarding.setFlow(newFlow);
+    push(`/onboarding/${newFlow[1].url}`);
   };
 
+  const form = useForm<z.infer<typeof roleSchema>>({
+    resolver: zodResolver(roleSchema),
+    defaultValues: {
+      roles: (onboarding
+        ? Array.from(onboarding.roles.values() || []).join(',')
+        : '') as RoleOptionType,
+    },
+  });
+
+  const { roles } = form.watch();
+
   return (
-    <>
-      <H1>Roles</H1>
-
-      <InputDropdown
-        label="What role(s) would you like to hold?"
-        multi
-        defaultValue={onboarding?.roles}
-        onChange={v => setRoles(v as Set<RoleEnum>)}
-        options={roleOptions}
-      />
-    </>
+    <FormProvider {...form}>
```

Comment:
Based on which role is selected, the div bounces up and down a little. See:

![Uploading Screenshot 2024-04-21 at 4.02.55 PM.png…]()

Reply:
I LOVE the description box btw

Reply:
yep eighthead should be fixed